lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group_metric_filter.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: cloudwatchlogs_log_group_metric_filter
+version_added: "2.9"
+short_description: Manage CloudWatch log group metric filter
+description:
+    - Create, modify and delete Cloudwatch log group metric filter.
+    - Cloudwatch log group metric filter can be use with I(ec2_metric_alarm).
+requirements:
+    - boto3
+    - botocore
+extends_documentation_fragment:
+    - aws
+author:
+    Markus Bergholz (@markuman)
+options:
+    state:
+        description:
+            - Whether the rule is present, absent or get
+        choices: ["present", "absent"]
+        default: present
+        required: false
+    log_group_name:
+        description:
+            - The name of the log group where the metric filter is applied on.
+        required: true
+    filter_name:
+        description:
+            - A name for the metric filter you create.
+        required: true
+    filter_patten:
+        description:
+            - A filter pattern for extracting metric data out of ingested log events.
+    metric_transformation:
+        description:
+            - A collection of information that defines how metric data gets emitted.
+        suboptions:
+            metricName:
+            metricNamespace:
+            matricValue:
+            defaultValue:
+'''
+
+EXAMPLES = '''
+- name: set metric filter on log group /fluentd/testcase
+    lekker_cloudwatch_metric_filter:
```

Comment:
is this supposed to be the module name?

Reply:
oh, of course not.  
but all cloudwatch* module naming as debatable.

strictly is must be `cloudwatch_log_metric_filter` and `cloudwatch_alarm` instead of `cloudwatchlogs_log_metric_filter` and `ec2_metric_alarm`, because the service/product is `cloudwatch`.